src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1033,19 +1026,11 @@ static void write_action_begin_locked(void* gt, grpc_error* error_ignored) {
     if (!t->is_first_write_in_batch) {
       GRPC_STATS_INC_HTTP2_WRITES_CONTINUED();
     }
-    grpc_closure_scheduler* scheduler =
-        write_scheduler(t, r.early_results_scheduled, r.partial);
-    if (scheduler != grpc_schedule_on_exec_ctx) {
-      GRPC_STATS_INC_HTTP2_WRITES_OFFLOADED();
-    }
-    set_write_state(
-        t,
-        r.partial ? GRPC_CHTTP2_WRITE_STATE_WRITING_WITH_MORE
-                  : GRPC_CHTTP2_WRITE_STATE_WRITING,
-        begin_writing_desc(r.partial, scheduler == grpc_schedule_on_exec_ctx));
-    GRPC_CLOSURE_SCHED(
-        GRPC_CLOSURE_INIT(&t->write_action, write_action, t, scheduler),
-        GRPC_ERROR_NONE);
+    set_write_state(t,
+                    r.partial ? GRPC_CHTTP2_WRITE_STATE_WRITING_WITH_MORE
+                              : GRPC_CHTTP2_WRITE_STATE_WRITING,
+                    begin_writing_desc(r.partial));
+    write_action(t, GRPC_ERROR_NONE);
```

Comment:
The issue over here would be that we would call grpc_endpoint_write while still holding on to the chttp2 combiner. Once I am done fixing the errors with https://github.com/grpc/grpc/pull/20331, I think this will be straight forward

Reply:
This is rebased on top of your PR. PTAL